src/internal/connector/onedrive/restore.go

```diff
@@ -599,3 +599,71 @@ func getMetadata(metar io.ReadCloser) (Metadata, error) {
 
 	return meta, nil
 }
+
+// Augment restore path to add extra files(meta) needed for restore as
+// well as do any other ordering operations on the paths
+func AugmentRestorePaths(backupVersion int, paths []path.Path) ([]path.Path, error) {
+	colPaths := map[string]path.Path{}
+	for _, p := range paths {
+		for {
+			np, err := p.Dir()
+			if err != nil {
+				return nil, err
+			}
+
+			onedrivePath, err := path.ToOneDrivePath(np)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(onedrivePath.Folders) == 0 {
+				break
+			}
+
+			_, found := colPaths[np.String()]
+			if !found {
+				colPaths[np.String()] = np
+			}
```

Comment:
Nit/curiosity: any reason not to let this clobber?
```suggestion
			colPaths[np.String()] = np
```

Reply:
You are right, it  doesn't matter. I think this is an artifact from me testing stuff out.